app/helpers/string_utility.rb

```diff
@@ -10,4 +10,11 @@ def ellipsize(
     mid_length = length - edge_length * 2
     string.gsub(/(#{edge}).{#{mid_length},}(#{edge})/, '\1...\2')
   end
+
+  def to_filesystems_compatible_filename(file_or_folder_name)
+    file_or_folder_name.strip
+                       .sub(/^[.-]*/, '')
+                       .sub(/\.*$/, '')
+                       .gsub(/[^\w',;. -]/, '_')
```

Comment:
This does not treat unicode chars well, since it works on ASCII, e.g. if I have a file `žščd.csv` it will be converted to `___d.csv`.